aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/DownloadTask.java

```diff
@@ -132,26 +132,30 @@ public Boolean call() {
             // because of a race condition in the network or OS.
             // interrupted and if its due to network drop, reset progress and
             // update state to WAITING_FOR_NETWORK.
-            if (RetryUtils.isInterrupted(e)) {
-                // Check if network is not connected, set the state to WAITING_FOR_NETWORK.
-                try {
-                    if (TransferNetworkLossHandler.getInstance() != null && 
-                        !TransferNetworkLossHandler.getInstance().isNetworkConnected()) {
-                        LOGGER.info("Thread:[" + Thread.currentThread().getId() + "]: Network wasn't available.");
-                        /*
-                         * Network connection is being interrupted. Moving the TransferState to
-                         * WAITING_FOR_NETWORK till the network availability resumes.
-                         */
-                        updater.updateState(download.id, TransferState.WAITING_FOR_NETWORK);
-                        LOGGER.debug("Network Connection Interrupted: " + "Moving the TransferState to WAITING_FOR_NETWORK");
-                        ProgressEvent resetEvent = new ProgressEvent(0);
-                        resetEvent.setEventCode(ProgressEvent.RESET_EVENT_CODE);
-                        progressListener.progressChanged(new ProgressEvent(0));
-                        return false;
-                    }
-                } catch (TransferUtilityException transferUtilityException) {
-                    LOGGER.error("TransferUtilityException: [" + transferUtilityException + "]");
+
+            // Check if network is not connected, set the state to WAITING_FOR_NETWORK.
+            try {
+                if (TransferNetworkLossHandler.getInstance() != null &&
+                    !TransferNetworkLossHandler.getInstance().isNetworkConnected()) {
+                    LOGGER.info("Thread:[" + Thread.currentThread().getId() + "]: Network wasn't available.");
+                    /*
+                     * Network connection is being interrupted. Moving the TransferState to
+                     * WAITING_FOR_NETWORK till the network availability resumes.
+                     */
+                    updater.updateState(download.id, TransferState.WAITING_FOR_NETWORK);
+                    LOGGER.debug("Network Connection Interrupted: " + "Moving the TransferState to WAITING_FOR_NETWORK");
+                    ProgressEvent resetEvent = new ProgressEvent(0);
+                    resetEvent.setEventCode(ProgressEvent.RESET_EVENT_CODE);
+                    progressListener.progressChanged(new ProgressEvent(0));
+                    return false;
                 }
+            } catch (TransferUtilityException transferUtilityException) {
+                LOGGER.error("TransferUtilityException: [" + transferUtilityException + "]");
```

Comment:
Control will flow to the RetryUtils.isInterrupted check if the TransferUtilityException is thrown. Is that intended?

Reply:
Yes, thats intended.